libyul/backends/evm/EVMDialect.h

```diff
@@ -61,15 +61,16 @@ struct BuiltinFunctionForEVM: BuiltinFunction
  */
 struct EVMDialect: public Dialect
 {
+	/// Constructor, should only be used internally. Use the factory functions below.
```

Comment:
Ah ok - it still can't be private because of the ``make_unique``, right? That's always a bit of a nuisance, but there's no really nice way around it...

Reply:
Yeah, I also find it really annoying...
Maybe it is possible to use something like `friend std::unique_ptr<ThisClass>`?

Reply:
That kind of defeats the purpose, though, isn't it? Then you can still use ``std::make_unique<ThisClass>`` outside the class...
There's a couple of "solutions", but none of them are really nice... one can locally inherit from ThisClass and ``make_unique`` the derived class, but that's ugly especially in the context of dynamic types... one can locally reimplement ``make_unique`` as
``
template<typename... Args>
std::unique_ptr<ThisClass> make_unique(Args&&... args)
{
    return std::unique_ptr<ThisClass>(new ThisClass(std::forward<Args>(args)...));
}
``
or one can just directly use ``std::unique_ptr<ThisClass>(new ThisClass(std::forward<Args>(args)...))`` in the factory function, if one is careful about exception safety... but none of that is really nice, so...

Reply:
There's not even a standard conformant, resp. guaranteed-to-work way to befriend make_unique, so...

Reply:
This would work :-):
```
#define private_make_unique(T) ([](auto&& ...args) { return std::unique_ptr<T>(new T(std::forward<decltype(args)>(args)...)); })

dialects[_version] = private_make_unique(EVMDialect)(AsmFlavour::Loose, false, _version);

```